home-assistant: add lifx component dependencies

netixx opened the pull request:
###### Motivation for this change
This commit solves https://github.com/NixOS/nixpkgs/issues/55527 by adding required
python packages and binding them in the component-packages.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

netixx commented:
Thank you @dotlambda for helping to debug my setup and making this PR possible !

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,28 @@
+{ stdenv,
```
```suggestion
{ lib
```

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,28 @@
+{ stdenv,
+  fetchPypi,
```
```suggestion
, fetchPypi
```

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,28 @@
+  // nix_run_setup: error: unrecognized arguments: test
+  doCheck = false;
+
+  nativeBuildInputs = [ ];
```
```suggestion
```

dotlambda commented on pkgs/development/python-modules/aiolifx_effects/default.nix:
```diff
@@ -0,0 +1,28 @@
+}:
+
+buildPythonPackage rec {
+  pname = "aiolifx_effects";
```
```suggestion
  pname = "aiolifx-effects";
```

netixx replied:
I originally did that, but I would get a 404 error when using a `-`. Did you have success with this syntax ?

dotlambda replied:
You have to add `pname = "aiolifx_effects";` inside `fetchPypi`.

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -168,6 +168,10 @@ in {
 
+  aiolifx = callPackage ../development/python-modules/aiolifx { };
+
+  aiolifx_effects = callPackage ../development/python-modules/aiolifx_effects { };
```
```suggestion
  aiolifx-effects = callPackage ../development/python-modules/aiolifx-effects { };
```
See https://nixos.org/nixpkgs/manual/#contributing.

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,28 @@
+  };
+
+  // tests are not implemented :
+  // nix_run_setup: error: unrecognized arguments: test
```
```suggestion
```

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,28 @@
+    sha256 = "cf53c9faea6eee25a466e73eef1753b82a75c7497648149c19c15342df2678f2";
+  };
+
+  // tests are not implemented :
```
```suggestion
  # tests are not implemented
```

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,28 @@
+  propagatedBuildInputs = [ bitstring ifaddr ];
+
+  meta = with stdenv.lib; {
+    homepage = "http://github.com/frawau/aiolifx";
```
```suggestion
    homepage = http://github.com/frawau/aiolifx;
```

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,28 @@
+  meta = with stdenv.lib; {
+    homepage = "http://github.com/frawau/aiolifx";
+    license = licenses.mit;
+    description = "API for local communication with LIFX devices over a LAN with asyncio.";
```
https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

netixx replied:
should I add myself to the maintainers in a separate (first) commit ?

dotlambda replied:
Yes, please.

dotlambda reviewed (changes requested):
Please make a separate commit for each package and add a `pythonPackages.` prefix to the commit messages.

dotlambda commented:
From https://github.com/frawau/aiolifx:
> aiolifx is a **Python 3**/asyncio library to control Lifx LED lightbulbs over your LAN.

netixx commented:
> From https://github.com/frawau/aiolifx:
> 
> > aiolifx is a **Python 3**/asyncio library to control Lifx LED lightbulbs over your LAN.

How do I transcribe this in the nix package manager ?

netixx committed:
pythonPackages.aiolifx: init at 0.6.7

Aiolifx package provides ability to control lifx (https://www.lifx.com)
light fixtures using python. The original need was to use it with
the home-assistant package, specifically the "lifx" component.

netixx committed:
pythonPackages.aiolifx-effects: init at 0.2.1

Aiolifx_effects package extends the pythonPackages.aiolifx package to
provide the ability to program effects (strobe, fade) into lifx light
fixtures (https://www.lifx.com) using python. The original need was to
use it with the home-assistant package, specifically the "lifx"
component. Although not strictly required to control the lights, the
lifx compopent imports this package and will fail if it's not present.

netixx commented:
Also, I included only `aiolifx_effects` (or `aiolifx-effects`) in the component dependencies, because `aiolifx` is a dependency, but the `lifx` components also uses the `aiolifx` directly (i.e. there are `import aiolifx` statements) so in a way it also has a direct dependency.

Should I add `aiolifx` in the component dependencies as well ?

dotlambda commented:
> > From https://github.com/frawau/aiolifx:
> > > aiolifx is a **Python 3**/asyncio library to control Lifx LED lightbulbs over your LAN.
> 
> How do I transcribe this in the nix package manager ?

`disabled = !isPy3k`

> Also, I included only `aiolifx_effects` (or `aiolifx-effects`) in the component dependencies, because `aiolifx` is a dependency, but the `lifx` components also uses the `aiolifx` directly (i.e. there are `import aiolifx` statements) so in a way it also has a direct dependency.
> 
> Should I add `aiolifx` in the component dependencies as well ?

Please read the first two lines of `component-packages.nix`. You need to run `parse-requirements.py`.

netixx committed:
maintainers: added netixx

netixx commented:
I fixed what you suggested.

I have a chicken and egg problem with running `parse-requirements.py` because I need the setup with the pythonPackages commit (and my current setup already runs on a custom branch).

Do you have a magic onliner (or few liners) that I can use to quickly generate the component-packages.nix ?

dotlambda commented:
I don't get what your problem is. Once you fixed all evaluation errors, there should be no problem running the script from your `hass-lifx-fix` branch.

dotlambda commented on pkgs/top-level/python-packages.nix:
```diff
@@ -299,7 +303,7 @@ in {
 
   cfn-flip = callPackage ../development/python-modules/cfn-flip { };
- 
+
```
please don't change random lines

netixx replied:
sorry my editor did this automatically when I saved the file (whitespace).

dotlambda commented on pkgs/development/python-modules/aiolifx-effects/default.nix:
```diff
@@ -0,0 +1,31 @@
+  # tests are not implemented
+  doCheck = false;
+
+  disabled = !isPy3k
```
;

netixx replied:
sorry for being such a drag... I missed this during all my rebasing...

dotlambda commented on pkgs/development/python-modules/aiolifx/default.nix:
```diff
@@ -0,0 +1,31 @@
+  # tests are not implemented
+  doCheck = false;
+
+  disabled = !isPy3k
```
;

netixx committed:
home-assistant: fix dependencies for lifx component

Update components dependencies for home-assistant.

netixx commented:
I wasn't sure it was going to work on my mac, but it did! However, I ran the script and another dependency was updated (`sensor.shodan`). I committed both for consistency but I will remove it if you think it's more appropriate.

dotlambda commented on pkgs/development/python-modules/aiolifx-effects/default.nix:
```diff
@@ -0,0 +1,31 @@
+{ lib
+  ,fetchPypi
```
```suggestion
, fetchPypi
```

netixx replied:
done (for all other as well).

dotlambda commented on pkgs/development/python-modules/aiolifx-effects/default.nix:
```diff
@@ -0,0 +1,31 @@
+  meta = with lib; {
+    homepage = https://github.com/amelchio/aiolifx_effects;
+    license = licenses.mit;
+    description = "aiolifx light effects";
```
Please be more descriptive and follow https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes.

netixx replied:
I used the description on the github to provide more compete explanation.

dotlambda commented:
> I committed both for consistency

That's fine with me.

dotlambda commented:
@GrahamcOfBorg build python3.pkgs.aiolifx-effects

netixx commented:
Thank you for your help doing this !
